lisp/transient.el

```diff
@@ -2882,6 +2886,14 @@ The last value is \"don't use any of these switches\"."
         (cadr (member value choices))
       (car choices))))
 
+(cl-defmethod transient-infix-read :around ((obj transient-dynamic-option))
+  "Populate object choices dynamically."
+  (with-slots (choices-function) obj
+    (let ((choices (funcall choices-function)))
+      (setf (oref obj choices) choices)
```

Comment:
elisp question, why setf and not just oset?

Reply:
It doesn't make a difference but since we use `oset` elsewhere, it should be used here as well.